### PR TITLE
8051 IL fixes

### PR DESCRIPTION
**Pull request by b1llow**
**Your checklist for this pull request**
- [ ] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [ ] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

The fixes I made when doing trace tests, using a modified [emu8051](https://github.com/imbillow/emu8051_bapframes)

**Test plan**

...

**Closing issues**

...

### CODE REVIEW

File: librz/analysis/arch/8051/8051_il.c [hunk 1]
```diff
@@ -1,11 +1,66 @@
 // SPDX-FileCopyrightText: 2021 Basstorm <basstorm@nyist.edu.cn>
-// SPDX-FileCopyrightText: 2022 imbillow <billow.fun@gmail.com>
+// SPDX-FileCopyrightText: 2023 billow <billow.fun@gmail.com>
```
Comment (wargio) [inline librz/analysis/arch/8051/8051_il.c]:
```suggestion
// SPDX-FileCopyrightText: 2022-2023 billow <billow.fun@gmail.com>
```

File: librz/analysis/arch/8051/8051_il.c [hunk 2]
```diff
@@ -168,230 +270,233 @@ static RzILOpPure *borrowin(RzILOpBitVector *a, RzILOpBitVector *b, RzILOpBitVec
...
-	return setb_any(op->argv[0], true);
+	return i8051_addressing_set_bool(op->argv[0], true);
 }
 static RzILOpEffect *i_cpl(I8051Op *op) {
 	rz_return_val_if_fail(op, NULL);
-	RzILOpPure *v = (op->argv[0]->mode == I8051_ADDRESSING_REGISTER && i8051_register_is_bit[op->argv[0]->d.reg]) || op->argv[0]->mode == I8051_ADDRESSING_BIT
-		? INV(get_any(op->argv[0]))
-		: LOGNOT(get_any(op->argv[0]));
-	return set_any(op->argv[0], v);
+	RzILOpPure *v = (op->argv[0]->mode == I8051_ADDRESSING_REGISTER && i8051_reg_is_psw_bit(op->argv[0]->d.reg)) || op->argv[0]->mode == I8051_ADDRESSING_BIT
```
Comment (XVilka) [inline librz/analysis/arch/8051/8051_il.c]: Usually, for such long conditions I recommend creating a separate `static inline` function with the descriptive na,e, which will be substituted by the compiler, see [DEVELOPERS.md](https://github.com/rizinorg/rizin/blob/dev/DEVELOPERS.md).

**Review by XVilka — CHANGES_REQUESTED** [review]
As long as it passes the tracetest, looks good.

**Comment by XVilka** [comment]
Please also send a PR to `rz-tracetest`, and if the modifications of `8051emu` are ready - transfer the repo to `rizinorg/` organization.

**Comment by b1llow** [comment]
https://github.com/rizinorg/rz-tracetest/pull/9

A PR for 8051 Trace Adapter

https://github.com/rizinorg/emu8051

This modified `emu8051` has also been transferred to `rizinorg`

**Comment by b1llow** [comment]
I haven't tested much of the 8051 code, so the tests I've done may not be very comprehensive.

**Review by XVilka — APPROVED** [review]

**Review by wargio — APPROVED** [review]

**Pull request merged**